persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/Conversion.java

```diff
@@ -74,6 +76,54 @@
 public class Conversion {
   private static final Logger LOG = LoggerFactory.getLogger(Conversion.class);
 
+  // A number of constructors for classes related to QueryOptions but that are not accessible in C*
+  // at the moment and need to be accessed through reflection.
+
+  // SpecificOptions(int pageSize, PagingState state, ConsistencyLevel serialConsistency, long
+  // timestamp)
+  private static final Constructor<?> specificOptionsCtor;
+  // DefaultQueryOptions(ConsistencyLevel consistency, List<ByteBuffer> values, boolean
+  // skipMetadata, QueryOptions.SpecificOptions options, ProtocolVersion protocolVersion)
+  private static final Constructor<?> defaultOptionsCtor;
+  // OptionsWithNames(QueryOptions.DefaultQueryOptions wrapped, List<String> names)
+  private static final Constructor<?> optionsWithNameCtor;
+
+  static {
+    try {
+      Class<?> defaultOptionsClass =
+          Class.forName("org.apache.cassandra.cql3.QueryOptions$DefaultQueryOptions");
```

Comment:
nit: would it be possible to declare this as `Class<org.apache.cassandra.cql3.QueryOptions>` to avoid type casts when we invoke this contructor?

Reply:
Well, unfortunately no. The point here is that the `QueryOptions.DefaultQueryOptions` which we want to construct is not visible.